consul/agents.go

```diff
@@ -47,18 +51,45 @@ func (a *ConcurrentAgents) getRandomAgentHost() string {
 	return hosts[idx]
 }
 
-func (a *ConcurrentAgents) GetAgent(agentHost string) (*consulapi.Client, error) {
+func (a *ConcurrentAgents) RemoveAgent(agentAddress string) {
 	a.lock.Lock()
 	defer a.lock.Unlock()
-	if agent, ok := a.agents[agentHost]; ok {
+
+	if IP, err := utils.HostToIPv4(agentAddress); err != nil {
+		log.WithError(err).Error("Could not remove agent from cache")
+	} else {
+		agentAddress = IP.String()
+		if _, ok := a.agents[agentAddress]; ok {
```

Comment:
Yep, I know, but we had the "decrement gauge" metric being updated before, so it was intended. Now, this can be simplified, as we always pass the actual cache size on update.